prov/efa/src/rdm/rxr_ep.c

```diff
@@ -1117,6 +1120,47 @@ static int rxr_ep_set_cuda_api_permitted(struct rxr_ep *ep, bool cuda_api_permit
 	return 0;
 }
 
+/**
+ * @brief set sendrecv_in_order_aligned_128_bytes flag in rxr_ep
+ * Supporting send/receive aligned 128 bytes buffer in order is more complicated than
+ * supporting RDMA write. There is no plan to support it in the near future.
+ * Therefore this function always returns -FI_EOPNOTSUPP if the user tries to enable it.
+ *
+ * @param[in,out]	ep					endpoint
+ * @param[in]		sendrecv_in_order_aligned_128_bytes	whether to enable in_order send/recv
+ *								for each 128 bytes aligned buffer
+ * @return		0 on success, -FI_EOPNOTSUPP if the option cannot be supported
+ */
+static
+int rxr_ep_set_sendrecv_in_order_aligned_128_bytes(struct rxr_ep *ep,
+						   bool sendrecv_in_order_aligned_128_bytes)
+{
+	if (sendrecv_in_order_aligned_128_bytes)
+		return -FI_EOPNOTSUPP;
+
+	ep->sendrecv_in_order_aligned_128_bytes = sendrecv_in_order_aligned_128_bytes;
+	return 0;
+}
+
+/**
+ * @brief set write_in_order_aligned_128_bytes flag in rxr_ep
+ * @param[in,out]	ep					endpoint
+ * @param[in]		write_in_order_aligned_128_bytes	whether to enable RDMA in order write
+ *								for each 128 bytes aligned buffer.
+ * @return		0 on success, -FI_EOPNOTSUPP if the option cannot be supported.
+ */
+static
+int rxr_ep_set_write_in_order_aligned_128_bytes(struct rxr_ep *ep,
+						bool write_in_order_aligned_128_bytes)
+{
+	if (write_in_order_aligned_128_bytes &&
+	    !efa_base_ep_support_op_in_order_aligned_128_bytes(&ep->base_ep, IBV_WR_RDMA_WRITE))
+		return -FI_EOPNOTSUPP;
```

Comment:
Is this correct? If the user sets`write_in_order_aligned_128_bytes=0`, should we ever return `-FI_EOPNOTSUPP`?